Docker Container Escape via runC overwrite with RCE as root (CVE-2019-5736) (#15107)
rapid7/metasploit-framework: documentation/modules/exploit/linux/local/docker_runc_escape.md

version [`18.09.2`][3].

This module has been successfully tested on Ubuntu 18.04.5 x64 and Fedora 28
x64. However, it doesn't seem to work on CentOS 7 x64. Also, it looks like the
exploit is more reliable on Fedora than Ubuntu.

### Installation

#### Ubuntu 18.04.5 x64 with Docker version 18.03.1
```
curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
echo "deb [arch=amd64 signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
apt-get update
apt-get install docker-ce=18.03.1~ce~3-0~ubuntu
```

#### Fedora 28 x64 with Docker version 18.03.1
```
dnf remove docker docker-client docker-client-latest docker-common docker-latest docker-latest-logrotate docker-logrotate docker-selinux docker-engine-selinux docker-engine
dnf -y install dnf-plugins-core
dnf config-manager --add-repo https://download.docker.com/linux/fedora/docker-ce.repo
dnf -y install docker-ce-18.03.1.ce-3.fc28
```

### Side Effects

#### runc
The host `runc` binary will be overwritten during exploitation. The module

> takes care of making a backup before the overwrite and restore it when the new

Reviewer: ```suggestion
takes care of making a backup before the overwrite and restoring it when the new
```

session is established. However, it might not work as expected and something

> could go wrong during the exploitation, which might avoid the session being

Reviewer: ```suggestion
could go wrong during the exploitation, which might prevent the session being
```

created. In this case, `runc` won't be restored and the the host will no longer
be able to run Docker containers. This process will need to be done manually
somehow by following the instruction displayed during the module execution:
```
cp <path to docker-runc backup> <path to docker-runc>
```

#### shell
The shell binary inside the container (set by the `OVERWRITE` option) will also

> be overwritten. However, the module makes a backup prior the overwrite and

Reviewer: ```suggestion
be overwritten. However, the module makes a backup priorr to the overwrite and
```
